pkg/build/strategies/sti/sti.go

```diff
@@ -317,6 +317,17 @@ func (builder *STI) Prepare(config *api.Config) error {
 				}
 			}
 		}
+
+		if len(config.AssembleRuntimeUser) == 0 {
+			if config.AssembleRuntimeUser, err = builder.docker.GetAssembleRuntimeUser(config.RuntimeImage); err != nil {
+				builder.result.BuildInfo.FailureReason = utilstatus.NewFailureReason(
+					utilstatus.ReasonInvalidArtifactsMapping,
+					utilstatus.ReasonMessageInvalidArtifactsMapping,
```

Comment:
Also this will cause failures for existing users who do not have this set in their runtime image.  You need to fall back to the previous behavior(uses the default user of the image i believe) if you can't determine an assemble-user from the command line args + image labels.

Reply:
Ok, thank you for taking your time for the review. Now *--assemble-runtime-user* is optional.